src/cpu/aarch64/acl_reorder.hpp

```diff
@@ -121,43 +145,32 @@ struct acl_reorder_fwd_t : public primitive_t {
             ACL_CHECK_SUPPORT(format_tag::undef == src_tag,
                     "Only ab, ba or cdba source formats supported");
 
-            auto dst_tag = memory_desc_matches_one_of_tag(*dst_md,
-                    format_tag::BA8b4a, format_tag::BA4b4a, format_tag::Ab4a,
-                    format_tag::Ab8a, format_tag::Acdb8a, format_tag::Acdb4a);
-            ACL_CHECK_SUPPORT(format_tag::undef == dst_tag,
-                    "Only Ab4a/Ab8a, BA8b4a/BA4b4a and Acdb8a/Acdb4a "
-                    "destination formats supported");
-
-            if (dst_tag == format_tag::BA4b4a || dst_tag == format_tag::Acdb4a
-                    || dst_tag == format_tag::Ab4a) {
-                _pd->app_.dst_wf = arm_compute::WeightFormat::OHWIo4;
-            } else if (mayiuse(sve_256)
-                    && (dst_tag == format_tag::BA8b4a
-                            || dst_tag == format_tag::Acdb8a
-                            || dst_tag == format_tag::Ab8a)) {
-                _pd->app_.dst_wf = arm_compute::WeightFormat::OHWIo8;
-            } else {
-                return status::unimplemented;
-            }
+            int transpose = do_transpose(src_md, dst_md);
+            ACL_CHECK_SUPPORT(transpose == -1, "Reorder not supported by ACL");
+            _pd->app_.transpose = (bool) transpose;
+
+            ACL_CHECK_SUPPORT(dst_md->format_desc.blocking.inner_nblks > 2, "destination format not supported");
+            
+            _pd->app_.dst_wf = arm_compute::WeightFormat::OHWI;
```

Comment:
You might want to make a temp references to make some of this logic look cleaner. E.g.
```
auto& dst_wf = _pd->app_.dst_wf;
auto& dst_format = dst_md->format_desc.blocking;
```